app/javascript/githubRepos/githubRepos.jsx

```diff
@@ -0,0 +1,60 @@
+import { h, Component } from 'preact';
+import { SingleRepo } from './singleRepo';
+
+export class GithubRepos extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      repos: [],
+      erroredOut: false,
+    };
+  }
+
+  componentDidMount() {
+    this.getGithubRepos();
+  }
+
+  getGithubRepos = () => {
+    fetch(`/api/github_repos`, {
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      credentials: 'same-origin',
+    })
+      .then(response => response.json())
+      .then(json => {
+        this.setState({ repos: json });
+      })
+      .catch(error => {
+        console.log(error);
+        this.setState({ erroredOut: true });
+      });
+  };
+
+  render() {
+    const { repos, erroredOut } = this.state;
+    const allRepos = repos.map(repo => (
+      <SingleRepo
+        githubIdCode={repo.github_id_code}
+        name={repo.name}
+        fork={repo.fork}
+        selected={repo.selected}
+      />
+    ));
+
+    if (erroredOut) {
+      return (
+        <div className="github-repos github-repos-errored">
+          An error occurred. Please check your browser console and email
```

Comment:
This adds error handling to the page, in case the asynchronous route times out or some other error occurs.